app/src/main/java/com/example/pomodorotimer/MainActivity.kt

```diff
@@ -268,13 +279,38 @@ class MainActivity : AppCompatActivity() {
             return
         }
 
-
-
         timer.needResume = false
         timer.isCounting = true
         startCountDownIntent.putExtra("toCount", timer.toSeconds())
         startService(startCountDownIntent)
     }
 
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putLong("timeLeftInSecond",timer.toSeconds())
+        outState.putInt("workState",timer.workState.ordinal)
+        outState.putBoolean("isCounting",timer.isCounting)
+        outState.putBoolean("needResume",timer.needResume)
+        outState.putInt("workTimer",timer.workTimer)
+        outState.putInt("breakTimer",timer.breakTimer)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+
+        Log.i("Restore Instance", "timeLeftInSeconds = ${timer.toSeconds()}")
+        timer.restoreFromSeconds(savedInstanceState.getLong("timeLeftInSecond"))
+        timer.workState = WorkState.getValueFromInt(savedInstanceState.getInt("workState"))
+        timer.needResume = savedInstanceState.getBoolean("needResume")
+        timer.isCounting = savedInstanceState.getBoolean("isCounting")
+        timer.workTimer = savedInstanceState.getInt("workTimer")
+        timer.breakTimer = savedInstanceState.getInt("breakTimer")
+
+        val startCountDownIntent = Intent(this, CountDownService::class.java)
+        startCountDownIntent.putExtra("toCount", timer.toSeconds())
+        startService(startCountDownIntent)
```

Comment:
if timer was counting before the orientation change, (notice service is also running in the background), and on orientation resume, we call startservice again (would that create 2 services)?

I was thinking 
either
 destroy the service on orientation change (like how we handled pause), and resume (start a new timer with the remaining time)
or
 leave the service running as it is (like the way we have right now), and no need to start timer again (since the last one is already running)

Not sure about the service behavior on orientation change, we need to do some testing.

Reply:
I thought the service was destroyed on orientation change, thats why the timer stopped